src/header/common/if_match.rs

```diff
@@ -0,0 +1,88 @@
+use header::{EntityTag, Header, HeaderFormat};
+use header::parsing;
+use std::fmt;
+use std::str::FromStr;
+
+/// An entity tag match specification
+#[derive(PartialEq, Clone, Debug)]
+pub enum EntityTagMatch{
+    EntityTags(Vec<EntityTag>),
+    Star
+}
+
+/// The `If-Match` header
+///
+/// The `If-Match` request-header field is used with a method to make
+/// it conditional.  The client provides a list of entity tags, and
+/// the request is only executed if one of those tags matches the
+/// current entity.
+///
+/// See http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.24
+#[derive(Clone, PartialEq, Debug)]
+pub struct IfMatch(pub EntityTagMatch);
```

Comment:
I added the wrapper with the thought that it could be shared between `IfNoneMatch` and `IfMatch`, and to have something to implement a function on for checking if an entity matched.  Since `IfNoneMatch` and `IfMatch` need to use different versions of entity tag matching anyway (weak vs strong), maybe that is less important.